docs: non-finite-provider-push DMA walkthrough

correiaafonso12 opened the pull request:
## WHAT

Adds documentation for the management API walkthrough on how to use the non-finite provider push feature

## WHY

This feature included a new DMA endpoint and overall a new mechanism to transfer data. Documentation should follow to ease its adoption.

## FURTHER NOTES

Closes https://github.com/eclipse-tractusx/tractusx-edc/issues/2104

lgblaumeiser reviewed (changes requested):
In principle:
- Do not document the delta, but assume, that this is the only document read by someone who wants to use the feature. And do not assume, that he knows the existing push functionality or wants to understand the difference.
- Describe the whole impact to be considered on both sides provider and consumer. I would actually recommend to add some of the sequence diagrams from the discussions and issues to provide better insights. There was value in the discussions and we should not throw them away only because the issue is closed.

bmg13 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,62 @@
+To start a non-finite Provider-PUSH transfer simply initiate a Transfer Process with flow type PUSH to an non-finite 
+Asset. Once the transfer starts data will eventually arrive at the data destination, similarly to usual PUSH transfers.
+
+However, checking the Transfer Processes states after the transfer succeeded shows that they remain in the *STARTED* state, 
+unlike typical transfers which transition to *COMPLETED*. This indicates that the transfer is still ongoing and has not yet 
+concluded.
+
+At this stage, the Data Provider can initiate additional transfers on demand by invoking the `DataFlow` API. This API is
+available within the management API context and is exposed through the data plane. To trigger a new data transfer, execute 
+the following request:
```
instead of these two paragraphs
```
After the transfer has succeeded the Transfer Processes state is kept as *STARTED* instead of *COMPLETED*, since the transfer is still ongoing.
The Data Provider can initiate additional transfers on demand by invoking the `DataFlow` API. This API is
available within the management API context and is exposed through the data plane. To trigger a new data transfer, execute 
the following request:
```

bmg13 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,62 @@
+# Identifying Non Finite Data
+
+Assets can identify data sources that produce non-finite data. To do this, the Asset should be registered with the
+`isNonFinite` data address property set to `true`. For example, the following example request body registers an `HttpData` 
```
please mention this being an optional field

correiaafonso12 replied:
Added it at [e7e498c](https://github.com/eclipse-tractusx/tractusx-edc/pull/2138/commits/e7e498c36b7786756ba4ae6915e68283066f19bc)

correiaafonso12 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,62 @@
+While the **PULL** transfer model natively supports the transfer of non-finite data, **PUSH** transfers require special
+consideration.
+
+Typical PUSH transfer complete the Transfer Processes after sending the data that is currently available in the data 
```
Adapted to remove comparations with the standard PUSH

correiaafonso12 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,62 @@
+# Transferring Non-Finite Data
+
+To start a non-finite Provider-PUSH transfer simply initiate a Transfer Process with flow type PUSH to an non-finite 
+Asset. Once the transfer starts data will eventually arrive at the data destination, similarly to usual PUSH transfers.
```
I was using "eventually" as in "some point in the future", which I think it fits, but could be confusing. I've decided to remove it altogether, so it reads "... data from the data source will arrive at the data 
destination".

What happens if no data is available depends on the data source implementation. For example, if the data source is S3 and the desired object is not present in the bucket, the TP terminates with an error message. Each data source handles their own scenarios. I've not mention it directly as it's not something that this feature brings, but I can add it.

correiaafonso12 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,62 @@
+The Connector does not calculate the difference between what has been sent and the new data to transfer. All data present 
+on the data source at the time of the transfer will be pushed to the Consumer.
+
+To close the communications channel, either the Consumer or the Provider must manually terminate the Transfer Process.
```
I've added a brief section regarding credential rotation. I've decided to make it succinct as this walkthrough is already quite big as is, but I can add more detail if necessary.

correiaafonso12 commented:
@lgblaumeiser I've added sequence diagrams from the discussion. However, for that I added the .png files directly into the repo (please note the folder structure of the changes of this PR). As this is the only management API walkthrough that does this I'm not sure if these files should be present in the repo. If yes, should the .puml files also be present? If not, where could I add them?

lgblaumeiser reviewed (changes requested):
Hi @correiaafonso12 , we are close, just two formulation improvements, as I think my proposals are easier to understand.

correiaafonso12 commented on docs/usage/management-api-walkthrough/10_nonfiniteproviderpush/10_nonfiniteproviderpush.md:
```diff
@@ -0,0 +1,82 @@
+# Identifying Non Finite Data
+
+Assets can identify data sources that produce non-finite data by setting the `isNonFinite` property. This is an optional 
+data address property that represents the finiteness of the data address. Any value other than `true` will be ignored. 
```
True. Updated at 7f4086f
